### ISSUE TITLE
Diversity Metrics Evaluation as stand alone  

### ISSUE THREAD

**Issue by sabirdvd** [AnnikaLindh/Diverse_and_Specific_Image_Captioning#2]
Thank you for sharing this nice work. This work presented different metrics to measure caption diversity. Could you please share the code or point it me out how this metric can work as stand-alone for evaluating the diversity of two captions generated from two different models (on the same image). Thanks.

**Comment by AnnikaLindh** [comment]
Hi @sabirdvd ,
The code for the diversity metrics can be found in this file: **combined_model/evaluate_model/eval_stats.py**

I've used sqlite to store all captions generated by each model into a separate SQL table, and then I use SQL to calculate the number of distinct captions (for Diversity) and number of captions without an exact match in the full list of captions in the training data's table (for Novelty).

These metrics are meant to evaluate a model's overall statistics over the full test set to see how repetitive the captions it generates are over multiple examples, so you should generate captions over the full test set for each model (and you also need access to the training set's captions to compare to when calculating Novelty).

Let me know if you have any further questions!
